packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -620,10 +620,16 @@ - (FLTTextureMessage *)create:(FLTCreateMessage *)input error:(FlutterError **)e
     } else {
       assetPath = [_registrar lookupKeyForAsset:input.asset];
     }
-    player = [[FLTVideoPlayer alloc] initWithAsset:assetPath
-                                      frameUpdater:frameUpdater
-                                     playerFactory:_playerFactory];
-    return [self onPlayerSetup:player frameUpdater:frameUpdater];
+    @try {
+        player = [[FLTVideoPlayer alloc] initWithAsset:assetPath
+                                          frameUpdater:frameUpdater
+                                         playerFactory:_playerFactory];
+        return [self onPlayerSetup:player frameUpdater:frameUpdater];
+    }
+    @catch (NSException *exception) {
```

Comment:
did it crash before when the exception is thrown?

Reply:
for ios yes. using the code sample of https://github.com/flutter/flutter/issues/118660#issue-1537405692, it should crash and look like this

<img width="1395" alt="Screenshot 2023-06-28 at 9 47 56 PM" src="https://github.com/flutter/packages/assets/38291023/caf60791-acd5-41e0-bc26-0986601dd761">